buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-				Homepage: "Buildpack A Homepage",
-			},
-			Dir: filepath.Join(buildpacksDir, "A", "v1"),
-		}
-	})
+						descriptor.Buildpack.Version = "v1.clear"
+						descriptor.Dir = filepath.Join(storeDir, "A", "v1.clear")
+						descriptor.Buildpack.ClearEnv = true
+					})
 
-	it.After(func() {
-		os.RemoveAll(tmpDir)
-		mockCtrl.Finish()
-	})
+					it("provides a clear env", func() {
+						if _, err := descriptor.Build(buildpack.Plan{}, config, mockEnv); err != nil {
+							t.Fatalf("Error: %s\n", err)
+						}
+						if s := cmp.Diff(h.Rdfile(t, filepath.Join(appDir, "build-info-A-v1.clear")),
+							"TEST_ENV: cleared\n",
+						); s != "" {
+							t.Fatalf("Unexpected info:\n%s\n", s)
+						}
+					})
 
-	when("#Build", func() {
-		when("building succeeds", func() {
-			it.Before(func() {
-				mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-			})
+					it("sets CNB_ vars", func() {
+						if _, err := descriptor.Build(buildpack.Plan{}, config, mockEnv); err != nil {
+							t.Fatalf("Unexpected error:\n%s\n", err)
+						}
 
-			it("should ensure the buildpack's layers dir exists and process build layers", func() {
-				h.Mkdir(t,
-					filepath.Join(layersDir, "A"),
-					filepath.Join(appDir, "layers-A-v1", "layer1"),
-					filepath.Join(appDir, "layers-A-v1", "layer2"),
-					filepath.Join(appDir, "layers-A-v1", "layer3"),
-				)
-				h.Mkfile(t, "[types]\n  build = true",
-					filepath.Join(appDir, "layers-A-v1", "layer1.toml"),
-					filepath.Join(appDir, "layers-A-v1", "layer3.toml"),
-				)
-				// the testdata/buildpack/bin/build script copies the content of the appDir into the layersDir
-				gomock.InOrder(
-					mockEnv.EXPECT().AddRootDir(filepath.Join(layersDir, "A", "layer1")),
-					mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer1", "env"), env.ActionTypeOverride),
-					mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer1", "env.build"), env.ActionTypeOverride),
-				)
-				gomock.InOrder(
-					mockEnv.EXPECT().AddRootDir(filepath.Join(layersDir, "A", "layer3")),
-					mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer3", "env"), env.ActionTypeOverride),
-					mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer3", "env.build"), env.ActionTypeOverride),
-				)
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-				testExists(t,
-					filepath.Join(layersDir, "A"),
-				)
-			})
+						var actual string
+						t.Log("sets CNB_BUILDPACK_DIR")
+						actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-buildpack-dir-A-v1.clear"))
+						h.AssertEq(t, actual, descriptor.Dir)
 
-			it("should provide the platform dir", func() {
-				h.Mkfile(t, "some-data",
-					filepath.Join(platformDir, "env", "SOME_VAR"),
-				)
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L224-L234